liquid-staking/src/funds/delegate_vote.rs

```diff
@@ -0,0 +1,93 @@
+multiversx_sc::imports!();
+
+use crate::{
+    basics::{
+        constants::TEN_DAYS,
+        errors::{ERROR_BAD_PAYMENT_TOKEN, ERROR_LS_TOKEN_NOT_ISSUED},
+    },
+    contexts::base::StorageCache,
+    liquidity_pool,
+    setup::{self, vote::LockedFunds},
+};
+
+#[multiversx_sc::module]
+pub trait DelegateVoteModule:
+    setup::config::ConfigModule
+    + multiversx_sc_modules::default_issue_callbacks::DefaultIssueCallbacksModule
+    + liquidity_pool::LiquidityPoolModule
+    + setup::vote::VoteModule
+{
+    #[payable("*")]
+    #[endpoint(delegateVote)]
+    fn delegate_vote(&self, proposal: ManagedBuffer, vote_type: ManagedBuffer) {
+        self.blockchain().check_caller_is_user_account();
```

Comment:
Add proposal expiration check: if the proposal is still open ( if the current epoch is before the proposal’s end period). 
This would allow votes after intended end period.

Reply:
We don't actually care for this. This should be governance's job to deny an illegitimate vote.